### DISCUSSION TITLE
Best practices for turning a PyScaffold script into a modularized project

### DISCUSSION THREAD

**Discussion by wpd** [pyscaffold/pyscaffold#736] [Best Practices & Workflows]
I view PyScaffold as a demonstration of the best practices which should be followed when developing a Python application.

I wonder if you could provide any advice, (or perhaps change the Fibonacci example) that demonstrates the best practices as taught in a Python masters class regarding developing a script based application that is divided into modules.
I have found https://stackoverflow.com/questions/16981921/relative-imports-in-python-3, https://stackoverflow.com/questions/30669474/beyond-top-level-package-error-in-relative-import, and https://stackoverflow.com/questions/14132789/relative-imports-for-the-billionth-time, but am at an impasse regarding the best way to change the Fibonacci example into a script based application that divides its work into modules.

A very naive approach would be to move the `fib()` function into its own module and to add:
```
from module import fib
```
to the top of `skeleton.py`.  And that works fine when performing simple tests such as:
```
python src/subproject/skeleton.py 5
```
But, when I modify `setup.cfg` to install `fibonacci` as a `console_script`, and try to distribute that simple application as a wheel (`tox -e build; pip install wheel`), the `fibonacci` script fails to import the function from the module.

As I researched, this, I learned that this is a fairly common and complex issue.  Perhaps there is no "Best Practice" to follow.  But, if there is, I expect you PyScaffold developers would know it, and it would be nice if the documentation or example application could show that best practice.

So, I figured I'd ask.

--wpd

**Comment by Zaloog** [comment]
Hi @wpd  ,
I just finished my first project with pyscaffold, which id say is using a modularized approach.
You can check it out [here](https://github.com/Zaloog/kanban-python) (its a terminal kanban app btw)

Maybe this helps clearing some of your questions.
Feel free to ask questions, If you have any

**Comment by wpd** [comment]
Closing this out.  Blame it on brain fog.  Blame it on hacking late at night.  Blame it on relying on "it works when I run it in VSCode, but not when I install it on another computer/virtual environment". Blame it on poor note taking.  Of course, the correct answer (as @Zaloog implement in the terminal kanban app) is to:
```
from subproject.module import fib
```
and that works.
